packages/x-components/src/x-modules/history-queries/store/actions/update-history-queries-with-search-response.action.ts

```diff
@@ -29,18 +29,44 @@ export const updateHistoryQueriesWithSearchResponse: HistoryQueriesXStoreModule[
       if (indexOfHistoryQuery >= 0) {
         const historyQuery = state.historyQueries[indexOfHistoryQuery];
         const isCurrentSessionHistoryQuery = historyQuery.timestamp > state.sessionTimeStampInMs;
-        if (
-          !isCurrentSessionHistoryQuery ||
-          historyQuery.totalResults == null ||
-          historyQuery.totalResults < searchResponse.totalResults
-        ) {
+        let filters: Filter[] = [];
+        if (!isCurrentSessionHistoryQuery || historyQuery.totalResults == null || searchResponse) {
+          if (searchResponse.request.filters) {
+            filters = checkFilters(searchResponse.request.filters);
+          }
+
           const newHistoryQueries = state.historyQueries.slice();
           newHistoryQueries[indexOfHistoryQuery] = {
             ...historyQuery,
-            totalResults: searchResponse.totalResults
+            totalResults: searchResponse.totalResults,
+            filters: filters
           };
           return dispatch('setHistoryQueries', newHistoryQueries);
         }
       }
     }
   };
+
+/**
+ * Take filters from the request and push them into a list.
+ *
+ * @param requestFilters - Filters from the request.
+ *
+ * @returns A list of selected filters in the history query.
+ *
+ */
+function checkFilters(requestFilters: Record<string, Filter[]>): Filter[] {
+  const filtersApplied: Filter[] = [];
+  Object.keys(requestFilters).forEach(key => {
+    const facet = requestFilters[key];
+    facet.forEach(filter => {
+      if (filtersApplied.includes(filter)) {
+        const filterIndex = filtersApplied.findIndex(value => value.id === filter.id);
+        filtersApplied.splice(filterIndex, 1);
+      } else {
+        filtersApplied.push(filter);
+      }
+    });
+  });
```

Comment:
Here you're iterating the keys of the `requestFilters` records to then use it to retrieve the value, you can simplify and iterate the values directly
`Object.values(requestFilters).forEach(filter => {`

Also, if in the end you don't need to do the splice, you can try and make this with a `reduce`

````typescript
filtersApplied = Object.values(requestFilters).reduce((accFilters, filters) => {
  accFilters.push(...filters);
  return accFilters;
});
````